pam/pam.go

```diff
@@ -177,9 +180,24 @@ func getSocketPath(argc C.int, argv **C.char) string {
 	return socketPath
 }
 
-//export pam_sm_setcred
-func pam_sm_setcred(pamh *C.pam_handle_t, flags, argc C.int, argv **C.char) C.int {
-	return C.PAM_IGNORE
+// SetCred is the method that is invoked during pam_setcred request.
+func (h *pamModule) SetCred(pam.ModuleTransaction, pam.Flags, []string) error {
+	return pam.ErrIgnore
+}
+
+// ChangeAuthTok is the method that is invoked during pam_chauthtok request.
+func (h *pamModule) ChangeAuthTok(pam.ModuleTransaction, pam.Flags, []string) error {
+	return pam.ErrIgnore
+}
+
+// OpenSession is the method that is invoked during pam_open_session request.
+func (h *pamModule) OpenSession(pam.ModuleTransaction, pam.Flags, []string) error {
+	return pam.ErrIgnore
+}
+
+// CloseSession is the method that is invoked during pam_close_session request.
+func (h *pamModule) CloseSession(pam.ModuleTransaction, pam.Flags, []string) error {
+	return pam.ErrIgnore
```

Comment:
These are necessary to make the `pamModule` to implement the `pam.ModuleHandler` interface (as explained in `pamModule` doc).

I wanted to keep always this explicit and never provide a base "struct" in go-pam on purpose, so that there won't be unexpected behaviors and everything is controlled by the module implementation.

Now... Not sure this requires a comment, but let me know in case. If it does, though, I think it's a language failure ;-).